File: client/src/boundary/interactor/BoardD2BConverter.ts
import { Board } from "domain/index";
import { BoundaryBoard } from "boundary/model/index";
import { Converter } from "utils/Converter";
import { ShipD2BConverter } from "./ShipD2BConverter";

export class BoardD2BConverter extends Converter<Board, BoundaryBoard> {
  private shipD2BConverter: ShipD2BConverter;

  constructor(shipConverter: ShipD2BConverter) {
    super();
    this.shipD2BConverter = shipConverter;
  }

  convert(board: Board): BoundaryBoard {
    return new BoundaryBoard(
      board.grid,
      this.shipD2BConverter.convertAll(board.ships)
    );
  }
}

File: server/battleships/use_cases/implementation/src/main/java/eu/optas/use_cases/implementation/BoardD2BConverter.java
package eu.optas.use_cases.implementation;

import eu.optas.domain.Board;
import eu.optas.use_cases.api.BoundaryBoard;
import eu.optas.utils.Converter;

public class BoardD2BConverter extends Converter<Board, BoundaryBoard> {
    private final ShipD2BConverter shipD2BConverter;

    public BoardD2BConverter(ShipD2BConverter shipD2BConverter) {
        this.shipD2BConverter = shipD2BConverter;
    }

    @Override
    public BoundaryBoard convert(Board board) {
        return new BoundaryBoard(board.getGrid(), shipD2BConverter.convertAll(board.getShips()));
    }
}
